lib/logstash/filters/ruby.rb

```diff
@@ -34,7 +34,12 @@ def register
   def filter(event)
     return unless filter?(event)
 
-    @codeblock.call(event)
+    begin
+      @codeblock.call(event)
+    rescue Exception => e
```

Comment:
if there's an exception, `filter_matched` shouldn't be called. here's a failing spec:

```ruby
  describe "a rubyexception should not call filter_matched on the event" do

    config <<-CONFIG
      filter {
        ruby {
          code => "raise 'You shall not pass'"
          add_tag => ["ok"]
        }
      }
    CONFIG

    sample("message" => "hello world") do
      insist { subject["tags"] } == ["_rubyexception"]
    end
  end
```

I'd suggest moving filter_matched(event) to:

```ruby
begin
  @codeblock.call(event)
rescue # ..
  # ...
else # called if no exceptions were raised
  filter_matched(event)
end
```